trpcultivate_germplasm/src/Plugin/TripalImporter/GermplasmCrossImporter.php

```diff
@@ -0,0 +1,972 @@
+    $supported_file_extensions = $this->plugin_definition['file_types'];
+    $instance->setSupportedMimeTypes($supported_file_extensions);
+    $validators['file']['valid_data_file'] = $instance;
+
+    // -----------------------------------------------------
+    // Raw row level
+    // - File rows are properly delimited
+    $instance = $this->service_validatorPluginManager->createInstance('valid_delimited_file');
+    // Configure the number of columns in a single row for this validator. We
+    // want a minimum number of 6 columns, so no need to set strict.
+    $instance->setExpectedColumns(6, FALSE);
+    $this->expected_columns = $instance->getExpectedColumns();
+    // Set the MIME type of this input file.
+    $instance->setFileMimeType($file_mime_type);
+    $validators['raw-row']['valid_delimited_file'] = $instance;
+
+    // -----------------------------------------------------
+    // Header Level
+    // - All column headers match expected header format
+    $instance = $this->service_validatorPluginManager->createInstance('valid_headers');
+    // Use our $headers property to configure what we expect for a header in the
+    // input file.
+    $instance->setHeaders($this->headers);
+    // Configure the expected number of columns and set it to be strict.
+    $num_columns = count($this->headers);
+    $instance->setExpectedColumns($num_columns, TRUE);
+    $validators['header-row']['valid_header'] = $instance;
+
+    // -----------------------------------------------------
+    // Data Row Level
+    // - All data row cells in columns 0-5 are not empty
+    $instance = $this->service_validatorPluginManager->createInstance('empty_cell');
+    $indices = [
+      $header_index['Year'],
+      $header_index['Season'],
+      $header_index['Cross Number'],
+      $header_index['Uniquename'],
+      $header_index['Maternal Parent'],
+      $header_index['Paternal Parent'],
+      $header_index['Cross Type'],
+    ];
+    $instance->setIndices($indices);
+    $validators['data-row']['empty_cell'] = $instance;
+
+    // - The column 'Season' is one of: Winter, Spring, Summer, Fall
+    $instance = $this->service_validatorPluginManager->createInstance('value_in_list');
+    $instance->setIndices([$header_index['Season']]);
+    $instance->setValidValues($this->valid_seasons);
+    $validators['data-row']['valid_season'] = $instance;
+
+    // - Maternal Parent and Paternal Parent cells exist in the database.
+    $instance = $this->service_validatorPluginManager->createInstance('germplasm_name_exists');
+    // Set the logger since this validator uses a setter (setOrganismID)
+    // which may log messages.
+    $instance->setLogger($this->logger);
+    $indices = [
+      $header_index['Maternal Parent'],
+      $header_index['Paternal Parent'],
+    ];
+    $instance->setIndices($indices);
+    $organism_id = $form_values['organism'];
+    $instance->setOrganismID($organism_id);
```

Comment:
This requires the parents to be the same species as the children. I thought we had discussed using setGenus() for this validator to allow for interspecies crosses?

Reply:
I recall that we decided on requiring all crosses to be of the same species, but I completely forgot that the parents could be different 🙈 

Since whatever logic I use here would also be needed in the run() method in PR #37, I'm wondering if the most optimal way would be to create a `getGenus()` in the [Organism Trait](https://github.com/TripalCultivate/TripalCultivate/blob/4.x/src/TripalCultivateValidator/ValidatorTraits/Organism.php). This method could perform the lookup if only provided an organism ID from `setOrganismId()` and we could save the genus under a separate key in the `context` array to reuse later (as would the `setGenus()` method in addition to setting an array of organism IDs).